src/components/ContextMenuView/ChatContextMenuView.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import ContextMenu, {
+  ContextMenuAction,
+  ContextMenuOnPressNativeEvent
+} from 'react-native-context-menu-view';
+import {Alert, NativeSyntheticEvent} from 'react-native';
+
+import ChatSchema from '../../database/schema/ChatSchema';
+import ShareUtils from '../../utils/share';
+import useMessageHook from '../../hooks/core/chat/useMessageHook';
+import {COLORS} from '../../utils/theme';
+
+type BetterSocialContextMenuType = 'MyChatContextMenu' | 'TargetChatContextMenu';
+type BetterSocialContextMenuName = 'Copy' | 'Delete' | 'Reply';
+type BetterSocialContextMenuAction = ContextMenuAction & {title: BetterSocialContextMenuName};
+type BetterSocialContextMenuOnPressNativeEvent = ContextMenuOnPressNativeEvent & {
+  name: BetterSocialContextMenuName;
+};
+
+type ChatContextMenuViewProps = {
+  children: React.ReactNode;
+  contextMenuType: BetterSocialContextMenuType;
+  chat?: ChatSchema;
+};
+
+const MyChatContextMenu: BetterSocialContextMenuAction[] = [
+  {title: 'Copy', systemIcon: 'square.on.square'},
+  {title: 'Delete', systemIcon: 'trash', destructive: true}
+];
+
+const DeletedChatContextMenu: BetterSocialContextMenuAction[] = [];
+
+const TargetChatContextMenu: BetterSocialContextMenuAction[] = [
+  {title: 'Copy', systemIcon: 'square.on.square'}
+];
+
+const ChatContextMenuView = ({children, contextMenuType, chat}: ChatContextMenuViewProps) => {
+  const {copyMessage, deleteMessage} = useMessageHook(chat);
+
+  let contextMenuActions = MyChatContextMenu;
+  if (contextMenuType === 'TargetChatContextMenu') {
+    contextMenuActions = TargetChatContextMenu;
+  }
+
+  if (chat?.type === 'deleted') {
+    contextMenuActions = DeletedChatContextMenu;
+  }
+
+  const onContextMenuPressed = (e: NativeSyntheticEvent<ContextMenuOnPressNativeEvent>) => {
+    console.log('context menu pressed', chat?.message);
+    const event = e?.nativeEvent as BetterSocialContextMenuOnPressNativeEvent;
+
+    if (event.name === 'Copy') {
+      copyMessage();
+    } else if (event.name === 'Delete') {
+      console.log('delete chat', chat?.message);
+      Alert.alert('Delete Message?', '', [
+        {text: 'Cancel', style: 'cancel'},
+        {text: 'Delete', style: 'destructive', onPress: () => deleteMessage(0)}
+      ]);
+    }
+  };
+
+  return (
+    <ContextMenu
+      previewBackgroundColor={COLORS.transparent}
+      style={{flex: 1}}
+      actions={contextMenuActions}
+      onPress={onContextMenuPressed}>
+      {children}
+    </ContextMenu>
+  );
+};
+
+export default ChatContextMenuView;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `ChatContextMenuView` component seems well implemented. It uses the `useMessageHook` to get the `copyMessage` and `deleteMessage` functions, which are then used in the `onContextMenuPressed` function based on the action selected from the context menu. The context menu actions are also dynamically set based on the `contextMenuType` prop and the `chat.type`. 

However, there's a hardcoded value `0` passed to the `deleteMessage` function on line 59. If this is an index or identifier, it should be replaced with a dynamic value related to the specific message being deleted.

```diff
- {text: 'Delete', style: 'destructive', onPress: () => deleteMessage(0)}
+ {text: 'Delete', style: 'destructive', onPress: () => deleteMessage(chat.id)}
```

This assumes that `chat.id` is the correct identifier for the message to be deleted. Please replace `chat.id` with the appropriate identifier if it's not correct.

Also, consider adding error handling for the `copyMessage` and `deleteMessage` functions. If these functions fail (for example, due to network issues), the user should be notified.


<!-- This is an auto-generated comment by OSS CodeRabbit -->